frontend/src/employee-frontend/components/common/Filters.tsx

```diff
@@ -273,13 +274,16 @@ export const UnitFilter = React.memo(function UnitFilter({
     <>
       <Label>
         <LabelText>{i18n.filters.unit}</LabelText>
-        <ReactSelect
+        <Combobox
+          items={options}
           placeholder={i18n.filters.unitPlaceholder}
-          options={options}
-          onChange={(option) =>
-            option && 'id' in option ? select(option.id) : undefined
-          }
-          value={selected}
+          selectedItem={selected ?? null}
+          onChange={(option) => {
+            option && 'id' in option ? select(option.id) : select(null)
```

Comment:
There is no need to do the `'id' in option` check with `Combobox`, it was only mandatory with react select since it had crappy types.

Also, if you typed the `select` prop like this `select: (unit?: string) => void`, this could be simplified to 
```
onChange={(option) => {
  select(option.id)
}
```
and you wouldn't have to do `unit: id || undefined` in the `selectUnit` and `selectFinanceDecisionHandler` functions.